fields/types/text/TextType.js

```diff
@@ -37,7 +37,7 @@ text.prototype.addFilterToQuery = function (filter, query) {
 	query = query || {};
 	if (filter.mode === 'exactly' && !filter.value) {
 		query[this.path] = filter.inverted ? { $nin: ['', null] } : { $in: ['', null] };
-		return;
+		return query;
```

Comment:
@JedWatson I'm not sure but I think this was omitted on accident, changing `query` but then not returning it seems like a bug.

I added tests for this, but if this is the intended behaviour I'll happily revoke this change and update the tests accordingly.